src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,92 @@
+ * reading task data, and converting date strings into {@link LocalDate} objects.
+ * </p>
+ */
+public class Parser {
+    /**
+     * Parses the user input and returns its corresponding command type.
+     *
+     * @param input The user input to be parsed.
+     * @return The corresponding {@link CommandType} based on the user input.
+     */
+    public static CommandType parseCommand(String input) {
+        if (input.equalsIgnoreCase("bye")) {
+            return CommandType.BYE;
+        }
+        if (input.equalsIgnoreCase("list")) {
+            return CommandType.LIST;
+        }
+        if (input.startsWith("todo")) {
+            return CommandType.TODO;
+        }
+        if (input.startsWith("deadline")) {
+            return CommandType.DEADLINE;
+        }
+        if (input.startsWith("event")) {
+            return CommandType.EVENT;
+        }
+        if (input.startsWith("mark")) {
+            return CommandType.MARK;
+        }
+        if (input.startsWith("unmark")) {
+            return CommandType.UNMARK;
+        }
+        if (input.startsWith("delete")) {
+            return CommandType.DELETE;
+        }
+        if (input.startsWith("tasks on")) {
+            return CommandType.TASKS_ON_DATE;
+        }
+        if (input.startsWith("find")) {
+            return CommandType.FIND;
+        }
+        return CommandType.UNKNOWN;
+    }
+
+    /**
+     * Converts a date string from the user input into a {@link LocalDate} object.
+     *
+     * @param input The user input containing the date string.
+     * @return The {@link LocalDate} representation of the provided date string.
+     * @throws DukeException If the date format is invalid or there's an error in parsing.
+     */
+    public static LocalDate getLocalDate(String input) throws DukeException {
+        String[] dateParts = input.split(" ");
+        if (dateParts.length < 3) {
+            throw new DukeException("Please provide a valid date in the format d/M/yyyy.");
+        }
+        LocalDate givenDate;
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy");
+            givenDate = LocalDate.parse(dateParts[2], formatter);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Invalid date format. Please use d/M/yyyy.");
+        }
+        return givenDate;
+    }
+
+    /**
+     * Determines if a specific date falls within the start and end dates of an event.
+     *
+     * @param event The event to check against.
+     * @param date  The date to verify.
+     * @return {@code true} if the date is within the event's date range, otherwise {@code false}.
+     */
+    public static boolean isWithinEventDate(Event event, LocalDate date) {
+        LocalDate startDate = event.getFrom().toLocalDate();
+        LocalDate endDate = event.getTo().toLocalDate();
+        return (date.isEqual(startDate) || date.isEqual(endDate) || (
```

Comment:
Should this return statement be broken down into a few steps? It seems to me that this return statement is rather complicated and difficult to comprehend at a glance.